# Desafío Libre — Post Semana 08
**Curso:** Fundamentos de Python | **UMCA**
**Tipo:** Exploración autónoma — sin fases ni instrucciones numeradas

**Estudiante**: Kevin Nuñez Jaimes
---

## Las reglas del desafío libre

1. **No hay fases.** No hay "haz esto, luego esto".
2. **No hay salida esperada.** Tú decides qué preguntas hacerle a los datos.
3. **Usas las herramientas que quieras** — de cualquier semana.
4. **El docente no da pistas** durante los primeros 20 minutos.
5. Al terminar, cada equipo presenta **una conclusión** al grupo.

---

## El dataset

Tienes acceso al mismo archivo `datos_clinica.json` de la Semana 08.
15 pacientes con: cédula, nombre, edad, género, provincia, cantón,
enfermedades y medicamentos.

## La pregunta abierta

> **¿Qué historia cuentan estos datos sobre la salud de los pacientes?**

No hay una respuesta correcta. Hay muchas respuestas posibles.
Tu trabajo es encontrar **al menos tres** patrones o conclusiones
respaldadas por los datos.

In [3]:
import json

with open("clinica_s08_avanzado.json", "r", encoding="utf-8") as f:
    datos = json.load(f)

print(f"Dataset listo: {len(datos)} pacientes")
print(f"Campos disponibles: {list(datos[0].keys())}")
print()
print("--- Primer registro ---")
for k, v in datos[0].items():
    print(f"  {k}: {v}")

Dataset listo: 57 pacientes
Campos disponibles: ['carnet', 'nombre', 'edad', 'genero', 'provincia', 'canton', 'visitas', 'enfermedad', 'medicamento', 'activo']

--- Primer registro ---
  carnet: 117002159511
  nombre: Kevin Nuñez Jaimes
  edad: 24
  genero: M
  provincia: Heredia
  canton: Heredia
  visitas: 20
  enfermedad: estrés
  medicamento: futból
  activo: True


---
## Tu análisis

Usa las celdas de abajo para explorar los datos libremente.
Agrega las celdas que necesites (botón `+` en la barra de Jupyter).

**Preguntas que podrías hacerte (solo sugerencias):**
- ¿Hay diferencias en las enfermedades entre géneros?
- ¿Las provincias con más pacientes tienen enfermedades específicas?
- ¿Los pacientes mayores tienen más enfermedades que los jóvenes?
- ¿Hay medicamentos que siempre aparecen juntos?
- ¿Qué cantón tiene el promedio de edad más alto?
- ¿Cuál es la enfermedad más común en pacientes menores de 40?

In [4]:
# Tu código aquí — exploración libre
import json
import os
ARCHIVO = "clinica_s08_avanzado.json"
if os.path.exists(ARCHIVO):
    with open(ARCHIVO, "r", encoding="utf-8") as f:
        datos_clinica = json.load(f)
else:    datos_clinica = []
print(f"Dataset listo: {len(datos_clinica)} pacientes")
print(f"Campos disponibles: {list(datos_clinica[0].keys())}")
print()
print("--- Primer registro ---")
for k, v in datos_clinica[0].items():
    print(f"  {k}: {v}")

Dataset listo: 57 pacientes
Campos disponibles: ['carnet', 'nombre', 'edad', 'genero', 'provincia', 'canton', 'visitas', 'enfermedad', 'medicamento', 'activo']

--- Primer registro ---
  carnet: 117002159511
  nombre: Kevin Nuñez Jaimes
  edad: 24
  genero: M
  provincia: Heredia
  canton: Heredia
  visitas: 20
  enfermedad: estrés
  medicamento: futból
  activo: True


In [ ]:
# Sigue explorando...
# De esos 57 pacientes, ¿cuántos son hombres y cuántos mujeres? ¿Cuál es la edad promedio? 
hombres = 0
mujeres = 0
suma_edades = 0

for paciente in datos_clinica:
    suma_edades += paciente["edad"]
    if paciente["genero"] == "M":
        hombres += 1
    else:
        mujeres += 1
print("*" * 20)
print(" Cantidad de Pacientes por Género ")
print(f"Hombres: {hombres}")
print(f"Mujeres: {mujeres}")
print(f"Edad promedio: {suma_edades / len(datos_clinica):.2f}")

# ¿Cuáles son las enfermedades más comunes? 
enfermedades = {}
for paciente in datos_clinica:
    enfermedad = paciente["enfermedad"]
    enfermedades[enfermedad] = enfermedades.get(enfermedad, 0) + 1
print("*" * 20)
print("Enfermedades más comunes:")
for enfermedad, count in sorted(enfermedades.items(), key=lambda x: x[1], reverse=True):
    print(f"  {enfermedad}: {count}")

# ¿Qué medicamento se receta con más frecuencia? 
medicamentos = {}
for paciente in datos_clinica:
    medicamento = paciente["medicamento"]
    medicamentos[medicamento] = medicamentos.get(medicamento, 0) + 1
print("*" * 20)
print("Medicamento más recetado:")
for medicamento, count in sorted(medicamentos.items(), key=lambda x: x[1], reverse=True):
    print(f"  {medicamento}: {count}")

********************
 Cantidad de Pacientes por Género 
Hombres: 32
Mujeres: 25
Edad promedio: 36.02
********************
Enfermedades más comunes:
  estrés: 9
  alergia: 8
  migraña: 6
  gripe: 6
  hipertensión: 5
  sinusitis: 4
  gastritis: 3
  diabetes: 3
  lumbalgia: 3
  insomnio: 3
  artritis: 2
  asma: 2
  dermatitis: 2
  colesterol: 1
********************
Medicamento más recetado:
  acetaminofén: 14
  ibuprofeno: 10
  loratadina: 8
  losartán: 5
  amoxicilina: 4
  omeprazol: 3
  metformina: 3
  melatonina: 3
  salbutamol: 2
  hidrocortisona: 2
  futból: 1
  atorvastatina: 1
  naproxeno: 1


In [6]:
# Más análisis...
# 1. ¿Cuántas visitas han tenido en promedio?

total_visitas = sum(p["visitas"] for p in datos_clinica)
promedio_visitas = total_visitas / len(datos_clinica)

print("*" * 20)
print(f"Visitas promedio por paciente: {promedio_visitas:.2f}")

# 2. Paciente con MÁS visitas
paciente_mas_visitas = max(datos_clinica, key=lambda x: x["visitas"])
print("*" * 20)
print(f"Paciente con más visitas: {paciente_mas_visitas['nombre']} ({paciente_mas_visitas['visitas']} visitas)")

# 3. Paciente con MENOS visitas
paciente_menos_visitas = min(datos_clinica, key=lambda x: x["visitas"])
print("*" * 20)
print(f"Paciente con menos visitas: {paciente_menos_visitas['nombre']} ({paciente_menos_visitas['visitas']} visitas)")

# 4. Paciente más JOVEN
paciente_mas_joven = min(datos_clinica, key=lambda x: x["edad"])
print("*" * 20)
print(f"Paciente más joven: {paciente_mas_joven['nombre']} ({paciente_mas_joven['edad']} años)")

# 5. Paciente más VIEJO
paciente_mas_edad = max(datos_clinica, key=lambda x: x["edad"])
print("*" * 20)
print(f"Paciente de mayor edad: {paciente_mas_edad['nombre']} ({paciente_mas_edad['edad']} años)")


********************
Visitas promedio por paciente: 12.35
********************
Paciente con más visitas: Roberto González Pérez (25 visitas)
********************
Paciente con menos visitas: Laura Mora Castro (1 visitas)
********************
Paciente más joven: Roberto González Pérez (20 años)
********************
Paciente de mayor edad: Pablo Herrera Jiménez (60 años)


In [7]:

# ¿Cuántos pacientes tienen activo = true? ¿Cuántos pacientes tienen activo = false?
activos = sum(1 for paciente in datos_clinica if paciente["activo"])
inactivos = len(datos_clinica) - activos
print("*" * 20)
print(f"Pacientes activos: {activos}")
print(f"Pacientes inactivos: {inactivos}")

#Cantón de cada paciente
cantones = {}
for paciente in datos_clinica:
    canton = paciente["canton"]
    cantones[canton] = cantones.get(canton, 0) + 1
print("*" * 20)
print("Cantidad de pacientes por cantón:")
for canton, count in sorted(cantones.items(), key=lambda x: x[1], reverse=True):
    print(f"  {canton}: {count}")

********************
Pacientes activos: 51
Pacientes inactivos: 6
********************
Cantidad de pacientes por cantón:
  Escazú: 4
  Alajuela: 4
  Puntarenas: 4
  Liberia: 3
  Esparza: 3
  Palmares: 3
  Cañas: 3
  Heredia: 2
  San Ramón: 2
  Cartago: 2
  Limón: 2
  Siquirres: 2
  Moravia: 2
  Garabito: 2
  Talamanca: 2
  Tibás: 2
  Nicoya: 2
  Desamparados: 1
  Curridabat: 1
  Tres Ríos: 1
  San Pablo: 1
  Goicoechea: 1
  Santa Cruz: 1
  Osa: 1
  Matina: 1
  Santo Domingo: 1
  Barva: 1
  San Rafael: 1
  Grecia: 1
  La Unión: 1


In [ ]:
#más análisis...
def analizar_frecuencia(lista_datos, llave):
    conteo = {}
    for paciente in lista_datos:
        valor = paciente[llave]
        conteo[valor] = conteo.get(valor, 0) + 1
    return conteo

# 2. lista con las categorías que se quieren analizar
categorias = ["enfermedad", "medicamento", "canton"]

# 3. Bucle Anidado
# Un for que recorre las categorías y el otro (dentro de la función) recorre los pacientes
print("--- ANÁLISIS DE FRECUENCIA ---")
for cat in categorias:
    resultado = analizar_frecuencia(datos_clinica, cat)
    
    # Ordenamos y mostramos el top 3 de cada una
    top_3 = sorted(resultado.items(), key=lambda x: x[1], reverse=True)[:3]
    
    print(f"\nTop 3 en {cat.upper()}:")
    for nombre, cantidad in top_3:
        print(f" - {nombre}: {cantidad}")

--- ANÁLISIS DE FRECUENCIA ---

Top 3 en ENFERMEDAD:
 - estrés: 9
 - alergia: 8
 - migraña: 6

Top 3 en MEDICAMENTO:
 - acetaminofén: 14
 - ibuprofeno: 10
 - loratadina: 8

Top 3 en CANTON:
 - Escazú: 4
 - Alajuela: 4
 - Puntarenas: 4


---
## Mis conclusiones

Escribe aquí las conclusiones que encontraste:

**Conclusión 1:**
_..._
Hay 57 pacientes, de los cuales se posee información como carnet, nombre, edad, género, provincia, cantón, cantidad de visitas que reciben, nombre de enfermedad, medicamento recetado y si se encuentra activo o no.

**Conclusión 2:**
_..._
Las enfermedades más comunes son la de estrés con nueve registros y la más baja es el colesterol con un registro, además los medicamentos que más se recetan son acetaminofén con 14 registros y el menos recetado es naproxeno con 1 registro encontrado.

**Conclusión 3:**
_..._
Los pacientes tienen un valor de visita promedio equivalente a 12.35, donde el paciente más visitado corresponde al señor Roberto González Pérez (25 visitas), el menos visitado es la señora Laura Mora Castro con (1 visita). Además el paciente más joven es Roberto González Pérez con 20 años y el paciente con mayor edad es Pablo Herrera Jiménez con 60 años.
Finalmente, de los pacientes activos se tienen 51 y de pacientes inactivos hay 6 y cabe resaltar que también se calculan los cantones de donde proviene cada paciente para identificar su zona de procedencia.


**¿Qué herramientas de Python usé en este desafío?**
_(Marca las que usaste)_

- [ ] `for` + `if` (S06)
- [ ] `def` funciones (S06)
- [ ] `sorted()` con `key=` (S07)
- [ ] List comprehensions (S07)
- [ ] `zip()` (S07)
- [ ] Diccionarios `{}` (S08)
- [ ] `json.load()` (S08)
- [ ] Bucles anidados (S08)
- [ ] Otra: ___